### PR TITLE
IGNITE-18224 Sql. Allow hash output distribution for SET relations.

### PR DESCRIPTION
**Pull request by xtern**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedIntersect.java [hunk 1]
```diff
@@ -43,20 +43,20 @@ public IgniteSingleIntersect(
      * Constructor.
      * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedIntersect.java:44]: let's fix javadoc

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedMinus.java [hunk 2]
```diff
@@ -43,20 +43,20 @@ public IgniteSingleMinus(
      * Constructor.
      * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedMinus.java:44]: seems all javadoc for touched classes should be rewritten )

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/SetOpPlannerTest.java [hunk 3]
```diff
@@ -70,10 +74,48 @@ public void setup() {
...
+        createTable(publicSchema, "AFFINITY_TBL3", type,
+                IgniteDistributions.affinity(1, "Test3", "hash"));
+
+        createTable(publicSchema, "AFFINITY_TBL4", type,
+                IgniteDistributions.affinity(0, "Test4", "hash2"));
+    }
+
+    /**
+     * testSetOpAffinityNested
+     * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
```
[the remark this answers is not in the document]
Comment (xtern) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/SetOpPlannerTest.java]: Done

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedIntersect.java (hunk 1 above)
Comment (xtern) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedIntersect.java]: Done

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedMinus.java (hunk 2 above)
Comment (xtern) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteColocatedMinus.java]: Done

**Pull request merged**